### PR TITLE
ARROW-17205: [Dev][Release] Merge script should prompt for next version when maintenance branch is created

### PR DESCRIPTION
**Pull request by raulcd**
Currently the merge_pr script is asking to merge to 9.0.0 when the fetature freeze has been implemented. If the maintenance branch is created we should filter them from the possible versions.

### CODE REVIEW

**Comment by raulcd** [comment]
@kszucs @kou what do you think about this change on the merge script?

File: dev/merge_arrow_pr.py [hunk 1]
```diff
@@ -350,6 +362,11 @@ def is_merged(self):
     def is_mergeable(self):
         return bool(self._pr_data["mergeable"])
 
+    @property
+    def maintenance_branches(self):
+        return [x["name"] for x in self._github_api.get_branches()
+                if "maint-" in x["name"]]
```
Comment (kou) [inline dev/merge_arrow_pr.py:368]:
```suggestion
                if x["name"].startswith("maint-")]
```

**Review by kou — COMMENTED** [review]
I like this idea.

(We need to fix test failures before we merge this.)

**Comment by raulcd** [comment]
> (We need to fix test failures before we merge this.)

Thanks @kou! I just pushed the minor test fix + I've added a test for the filtering on the maintenance branches.

**Review by kszucs — APPROVED** [review]
LGTM

**Review by kou — APPROVED** [review]
+1

**Pull request merged**